File: test-jira.js
import dotenv from "dotenv";
import path from "path";
import { fileURLToPath } from "url";
import axios from "axios";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const envPath = path.join(__dirname, "jira-mcp-server", ".env");

console.log("Loading from:", envPath);
const result = dotenv.config({ path: envPath });

console.log("Environment loaded:", !!result.parsed);
console.log("JIRA_DOMAIN:", process.env.JIRA_DOMAIN);
console.log("JIRA_EMAIL:", process.env.JIRA_EMAIL);
console.log("Token length:", process.env.JIRA_API_TOKEN?.length);

const url = `${process.env.JIRA_DOMAIN}/rest/api/3/issue/KAN-4`;
console.log("URL to call:", url);

try {
  const response = await axios.get(url, {
    auth: {
      username: process.env.JIRA_EMAIL,
      password: process.env.JIRA_API_TOKEN
    }
  });
  console.log("Success! Response:", response.status);
} catch (error) {
  console.error("Error:", error.message);
  console.error("Code:", error.code);
  console.error("Response status:", error.response?.status);
}

File: test_jira.py
import os
import requests
from dotenv import load_dotenv

# Load env variables from orchestrator directory
orchestrator_dir = os.path.join(os.path.dirname(__file__), "orchestrator")
load_dotenv(os.path.join(orchestrator_dir, ".env"))

domain = os.getenv("JIRA_DOMAIN", "https://hpe-team2.atlassian.net")
email = os.getenv("JIRA_EMAIL")
token = os.getenv("JIRA_API_TOKEN")

if not email or not token:
    print("[-] Please configure JIRA_EMAIL and JIRA_API_TOKEN in your orchestrator/.env file!")
    exit(1)

url = f"{domain}/rest/api/3/issue/KAN-17"
print(f"[!] Fetching {url}...")

r = requests.get(
    url,
    auth=(email, token)
)

print(f"[+] Status Code: {r.status_code}")
print(r.text[:500])
